Changed processUserInput and keydown event listener

dcazrael opened the pull request.

dcazrael committed:
Update line 242 - 246

userInput wasn't passed as a parameter to processUserInput

- added parameter to processUserInput
- removed declaraction of userInput
- passed ev.key directly to processUserInput

dcazrael committed:
Merge pull request #1 from dcazrael/dcazrael-patch-1

Update line 242 - 246

dcazrael committed:
Fixing spelling mistake

dcazrael committed:
update entity generation

dcazrael committed:
update entity generation

dcazrael commented:
geotic v4 changed the way entities are created.

now you need a world container to create entities.

dcazrael committed:
additional fixes due to geotic v4 update

dcazrael committed:
part 5 complete

dcazrael committed:
adjusting for geotic v4 changes

dcazrael commented:
In part 4 when  checking if the entity has the property 'IsRevealed' we are actually looking for the class as per prototype description

So we don't want to put IsRevealed in quotations.

Also, please excuse the constant switching between single and double quotes. I had my editor not set for the correct workspace.

dcazrael committed:
reverted pre-emptive changes to userInput, further adjustment for geotic v4 changes

ddmills commented on tutorial/part3.md:
```diff
@@ -474,13 +474,13 @@ Finally in movement we need to check if the location we want to move to has an e
 +    const blockers = [];
-+    for (const e of ecs.entities.all) {
-+      if (e.position.x === mx && e.position.y === my && e.isBlocking) {
++    for (const e of world._entities.all) {
```
this can now use the helper method https://github.com/ddmills/geotic/blob/master/src/World.js#L22:

```
for (const e of world.getEntities()) {
    ...
}
```

dcazrael replied:
adjusted my pr

dcazrael committed:
adjusted ecs.entities.all to world.getEntities()

dcazrael committed:
adjust layerXxxEntitites.isMatch()  to layerXxxEntities.matches(), additional changes for geotic v4

dcazrael committed:
more adjustments relating to geotic v4 changes; fixed typos

dcazrael commented:
In a recent commit <Entity> and <EntityArray> have been removed from the code base of geotic.

Due to this regular arrays have to be used. This also means that destroy won't remove the entity from the array, but keep it as destroyed.
Due to this, I used a filter instead, which seems to have the expected result.

dcazrael committed:
adapt to geotic v4; fix spelling

dcazrael commented:
- reworked EntityArrays to work with geotic v4 based on ddmills recommendation
- fixed spelling mistakes
- added utility functions
- lots of adjustments for geotic v4

luetkemj commented:
@dcazrael thanks for all the work on this - I'm gonna try and review it this week/weekend.

dcazrael committed:
adjustments for geotic v4

dcazrael committed:
last adjustments for geotic v4; code cleanup

dcazrael commented:
No problem at all. Your code is an amazing starting point. I will try to continue where you left off and implement the rest of the TCOD tutorial. Thanks for the hard work.

dcazrael committed:
part 12

dcazrael committed:
part 13 - gearing up

dcazrael committed:
finish part 13

dcazrael committed:
Merge pull request #2 from dcazrael/part13

Part13

dcazrael committed:
fix readme
